src/disasm.cpp

```diff
@@ -484,28 +496,45 @@ void jl_strip_llvm_addrspaces(Module *m)
     PM.run(*m);
 }
 
+
+#include <iostream>
+#include "llvm/IR/InstIterator.h"
+#include <sstream>
 // print an llvm IR acquired from jl_get_llvmf
 // warning: this takes ownership of, and destroys, f->getParent()
 extern "C" JL_DLLEXPORT
 jl_value_t *jl_dump_function_ir_impl(void *f, char strip_ir_metadata, char dump_module, const char *debuginfo)
 {
     std::string code;
     raw_string_ostream stream(code);
+    std::vector<std::pair<int, std::string>> ccalls;
 
+    bool annotate_ccalls = strip_ir_metadata; // do not annotate when raw = true
     {
         Function *llvmf = dyn_cast_or_null<Function>((Function*)f);
+
         if (!llvmf || (!llvmf->isDeclaration() && !llvmf->getParent()))
             jl_error("jl_dump_function_ir: Expected Function* in a temporary Module");
-
+        int inst_count = 0;
+        if (annotate_ccalls) { // find all ccall instructions to annotate
+        for (inst_iterator I = inst_begin(llvmf), E = inst_end(llvmf); I != E; ++I) {
```

Comment:
Could we create this as part of `jl_strip_llvm_debug`?

Reply:
It can, 

done